File: obiektowe 2tiSP/Operatory_3SP/Operatory_3SP/Product.cpp
#include "Product.h"

#include <ostream>

Product::Product() {
	name = "noname";
	price = 0;
	description = "no description";
}

Product::Product(std::string name, double price, std::string description) {
	this->name = name;
	this->price = price;
	this->description = description;
}

std::ostream& operator<<(std::ostream& os, const Product& p) {
	os << "informace o produkcie:" << std::endl;
	os << "Nazwa produktu: " << p.name << std::endl;
	os << " Cena produktu: " << p.price << std::endl;
	os << " Opis produktu: " << p.description << std::endl;
	return os;
}

File: webowe2tiGim/WebMVC_selfDB/WebMVC_selfDB/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace WebMVC_selfDB.Models
{
    public class Product
    {
        [Key]
        public int Id { get; set; }

        [Display(Name ="Produkt")]
        [MaxLength(50,ErrorMessage = "Maksymalnie 50 znaków")]
        [Required]
        public string Name { get; set; }

        [Display(Name = "Cena")]
        [Required]
        [Range(0.00,Double.MaxValue,ErrorMessage = "Cena nie może być ujemna")]
        public decimal Price { get; set; }

        [Display(Name = "Opis")]
        [DataType(DataType.MultilineText)]
        public string Description { get; set; }

    }
}
